doc/source/development/rfc/rfc100_float16_support.rst

```diff
@@ -47,23 +52,33 @@ Motivation
 Details
 -------
 
-The following type aliases are defined in both C and C++, if supported
-by the compiler:
-- ``GFloat16`` --> e.g. ``_Float16``
+The type ``_Float16`` is defined in both C and C++, if supported by
+the compiler. This is the type name that C will (most likely) use in
+the future. Modern GCC releases provide this type already. OTher
+compilers might support this type under a different name, and in this
+case, GDAL will define ``_Float16`` as type alias.
```

Comment:
If we are not confident about _Float16 being the canonical name that will be universally adopted, then the GFloat16 approach would indeed be better (it would be dangerous for GDAL to define _Float16 as a global identifier starting with underscore is assumed to be reserved. cf https://en.cppreference.com/w/c/language/identifier).
But I'm not sure if there will actually be functions in the GDAL C API where GFloat16 will be exposed. This should be mostly an implementation detail

Reply:
At this point it's fairly certain that the type will be named `_Float16`. There is a draft addendum to the draft standard, and other types are called e.g. `_Decimal32`, so `_Float16` is a consistent choice. Of course the standard hasn't been adopted yet, so things might happen.

Reply:
The name is now `GFloat16`, mostly because using `_Float16` directly didn't work. (The standard C++ library doesn't support `isnan` or `numeric_limits` for `_Float16`.)